README.md

```diff
@@ -137,73 +112,80 @@ $ make
 If you also want to try the example Java benchmarks, you must build them
 as a separate step:
 
-```bash
+```sh
 $ pwd
 .../krun/examples/benchmarks
 $ make java-bench
 ```
 
 ## Step 5: Audit system services
 
-You should take some time to review the services running on your benchmarking
-machine. Debian especially has a habit of starting daemons which get pulled in
-by dependencies.
+Background services (e.g. mail servers or cron) can interfere with benchmarking.
+The more services that you are able to switch off, the less interference is
+likely to occur. Some services are best disabled at boot and/or permanently
+(depending on your OS) and must be done manually. However, you may wish
+to disable some services only during benchmarking (e.g. you may wish to have a mail server
+running before and after benchmarking to inform you of benchmarking progress),
+which can be specified in the `PRE_EXECUTION_CMDS` and `POST_EXECUTION_CMDS`
+settings in your Krun config file.
 
-Some services you can disable at boot. Others you may want disabled only for
-the duration of the benchmarking (e.g. mail servers, crond, atd, ntpd). For the
-latter kind, you can use `PRE_EXECUTION_CMDS` and `POST_EXECUTION_CMDS` in your
-Krun config file to stop and start the services.
-
-Note that by default Debian machines do not use a service like ntpd to set the
-system time. Instead the time is set using `ntpdate` when a network interface
-comes up.
 
 ### Linux
 
+Note that Debian has, from a benchmarking perspective, the unfortunate habit of
+automatically starting daemons which get pulled in by dependencies.
+
 On Linux, list services with:
 
-```
+```sh
 # systemctl | grep running
 ```
 
-Disable services (now and at boot) with:
+Permanently disable services (including after system reset) with:
 
-```
+```sh
 # systemctl stop <service>
 # systemctl disable <service>
 ```
 
-Commonly enabled services you probably don't want include:
+Commonly enabled services that you may wish to disable:
 
  * apache2
  * memcached
  * nfs-common
 
+
 ### OpenBSD
 
-On OpenBSD, list at services with:
+On OpenBSD, list services with:
 
-```
-# rcctl ls started
+```sh
+# doas rcctl ls started
 ```
 
-Disable services (now and at boot) with:
-```
+Permanently disable services (including after system reset) with:
+```sh
 # rcctl stop <service>
 # rcctl disable <service>
 ```
 
-Commonly enabled services you probably don't want include:
+Commonly enabled services that you may wish to disable:
 
  * pflogd
  * sndiod
 
+
 ## Step 6: Run the example experiment
 
-```bash
-$ cd ../
-$ pwd
-.../krun/examples
+The `examples` directory contains the `example.krun` experiment. This contains
+two benchmark programs (*nbody* and *dummy*), both of which are run on *C* and
+*CPython* (both of which we'll consider to be "VMs"). Each benchmark is run for
```

Comment:
PyPy not CPython.

Reply:
We don't ask people to install PyPy... Should we? Otherwise this will fail. It also feels really weird to have C as an example. Could we change it to PyPy and CPython maybe?

Reply:
I think we ditched CPython becuase we wanted two fast languages in the example which the test suite runs.

Reply:
Not sure why C isn't a valid example though.

Reply:
C compilers are ubiquitous, which makes it an example lots of people are likely to be able to run out of the box. PyPy, not so much, but CPython is slow.

Reply:
How slow is CPython? Is it bad enough to make this a big problem? If I'm a reader seeing "C" is pretty confusing -- confusing enough that the speed up probably may well not be worth it.

Reply:
The example runs nbody with a parameter of 15. Rough timings on my laptop:
```
wilfred:python> time pypy bench.py
pypy bench.py  4.08s user 0.17s system 93% cpu 4.536 total
wilfred:python> time python bench.py
python bench.py  64.40s user 19.94s system 99% cpu 1:25.03 total
```

I don't get why C is confusing. It's a language like any other than you can benchmark. What am I missing here? :P

Reply:
C isn't a VM and most people won't think that its warmup is relevant. The original text even said "VMs like PyPy and C"!

Are those timings for one iteration or a complete run of the benchmark?

Reply:
Is Krun just for warmup experiments then? I hoped it would be more general.

> Krun is a framework for running high-quality software benchmarking experiments.

The above timings are one pexec and we run two in the test suite.

Reply:
Point taken. Maybe I just need to rework the text to make it less confusing.